src/apns_connection.erl

```diff
@@ -201,7 +201,11 @@ handle_cast(Msg, State) when is_record(Msg, apns_msg) ->
         ok ->
           {noreply, State#state{out_expires = Timeout}};
       {error, Reason} ->
-        apns_queue:fail(State#state.queue, Msg#apns_msg.id),
+        InfoLoggerFun = State#state.info_logger_fun,
+        InfoLoggerFun("[ ~p ]ssl error:", [Socket]),
+        ssl:close(Socket),
+        handle_cast(Msg, State#state{out_socket = undefined}),
```

Comment:
You should check the response from `handle_cast` and update the `State` accordingly